src/components/BusinessSearchList/index.js

```diff
@@ -83,7 +85,7 @@ export const BusinessSearchList = (props) => {
     })
     setBusinessesSearchList({
       ...businessesSearchList,
-      businesses: updatedBusinesses
+      businesses: updatedBusinesses.filter(_business => _business?.city_id === cityId)
```

Comment:
Validate is cityId is different of undefined, because if you filter by undefined the result is zero